### PR TITLE
Delete `authentication.clientSecret` from HTTP responses

### PR DESCRIPTION
**Pull request by baoshan**
1. This PR deletes `authentication.clientSecret` from the following endpoints:

    1. create token
    2. create scoped token
    3. check token
    4. reset token
    5. refresh token

2. All above responses now have an `authentication` property. This is a breaking change for `create token`. 
3. All returned `authentication` properties now have `"type"` and `"tokenType"` set.

Thanks.

### CODE REVIEW

File: test/cloudflare-handler.test.ts [hunk 1]
```diff
@@ -109,10 +109,7 @@ describe("createCloudflareHandler(app)", () => {
   it("POST /api/github/oauth/token", async () => {
     const appMock = {
       createToken: jest.fn().mockResolvedValue({
-        authentication: {
-          token: "token123",
-          scopes: ["repo", "gist"],
-        },
+        authentication: { clientSecret: "" },
```
Comment (gr2m) [inline test/cloudflare-handler.test.ts:112]: why not leave the `token` and `scopes` properties in the test, only add `clientSecret`?

Comment (gr2m) [reply test/cloudflare-handler.test.ts:112]: Can you add `type: "token"` to `authentication` to test that we don't just set `authentication` to an empty object?

File: test/cloudflare-handler.test.ts [hunk 2]
```diff
@@ -145,7 +141,10 @@ describe("createCloudflareHandler(app)", () => {
 
   it("GET /api/github/oauth/token", async () => {
     const appMock = {
-      checkToken: jest.fn().mockResolvedValue({ id: 1 }),
+      checkToken: jest.fn().mockResolvedValue({
+        foo: "bar",
```
Comment (gr2m) [inline test/cloudflare-handler.test.ts:145]: I'd prefer not to add made up properties, it might be confusing to contributors and our future selfs. Can we set `data: { id: 1 }` instead of `foo: "bar"`?
```suggestion
        data: { id: 1 },
```

Comment (gr2m) [reply test/cloudflare-handler.test.ts:145]: could you do the same where you set `foo: "bar"`?

I can do it for you if you are busy, I really appreciate you taking on this important bug fix

File: test/cloudflare-handler.test.ts [hunk 3]
```diff
@@ -221,35 +221,26 @@ describe("createCloudflareHandler(app)", () => {
...
-    expect(await response.json()).toMatchInlineSnapshot(`
-      Object {
-        "authentication": Object {
-          "token": "scopedtoken456",
-        },
-        "data": Object {
-          "id": 1,
-        },
-      }
-    `);
```
Comment (gr2m) [inline test/cloudflare-handler.test.ts:233]: you can update the inline snapshots by running `jest` with `--updateSnapshot`

**Review by gr2m — COMMENTED** [review]
> The `authentication` property in the `create token` response has `"type"` and `"tokenType"` set while the others have not. I think this is not beautiful and will cascade that inconsistence to `auth-oauth-user-client.js`

that was definitely not intended.

> To main consistence, all above responses now have an `authentication` property. This is a breaking change for `create token`.

that's ok, I consider it a bug fix. It might cause some friction for people who used the other behavior, but I have to balance it with the friction it would cause to create a breaking release because of this. The usage of the OAuth middleware is very low right now.

File: test/cloudflare-handler.test.ts (hunk 1 above)
Comment (baoshan) [reply test/cloudflare-handler.test.ts]: Done.

File: test/cloudflare-handler.test.ts (hunk 2 above)
Comment (baoshan) [reply test/cloudflare-handler.test.ts]: Done.

File: test/cloudflare-handler.test.ts (hunk 3 above)
Comment (baoshan) [reply test/cloudflare-handler.test.ts]: Done.

**Comment by baoshan** [comment]
Please review the PR when you have time. I’ll continue working on `auth-oauth-user-client.js` later.

Thanks.

**Review by gr2m — APPROVED** [review]
great PR, thank you!!

**Comment by gr2m** [comment]
> 3. All returned `authentication` properties now have `"type"` and `"tokenType"` set

My understanding is that we should fix that in https://github.com/octokit/oauth-methods.js, and then remove the workaround in this repository again, right? I can make a follow up issue, just double checking with you first
